include/seqan3/alignment/aligned_sequence/debug_stream_alignment.hpp

```diff
@@ -0,0 +1,113 @@
+ * \tparam alignment_t  The type of the alignment, must satisfy tuple_like.
+ * \tparam idx          An index sequence.
+ * \param[in] stream    The output stream that receives the formatted alignment.
+ * \param[in] align     The alignment that shall be streamed.
+ */
+template <typename char_t, tuple_like alignment_t, size_t ...idx>
+void stream_alignment(debug_stream_type<char_t> & stream, alignment_t const & align, std::index_sequence<idx...> const & /**/)
+{
+    using std::get;
+    size_t const alignment_size = get<0>(align).size();
+
+    // split alignment into blocks of length 50 and loop over parts
+    for (size_t begin_pos = 0; begin_pos < alignment_size; begin_pos += 50)
+    {
+        size_t const end_pos = std::min(begin_pos + 50, alignment_size);
+
+        // write header line
+        if (begin_pos != 0)
+            stream << '\n';
+
+        stream << std::setw(7) << begin_pos << ' ';
+        for (size_t pos = begin_pos + 1; pos <= end_pos; ++pos)
+        {
+            if (pos % 10 == 0)
+                stream << ':';
+            else if (pos % 5 == 0)
+                stream << '.';
+            else
+                stream << ' ';
+        }
+
+        // write first sequence
+        stream << '\n' << std::setw(8) << "";
+        std::ranges::for_each(get<0>(align) | views::slice(begin_pos, end_pos) | views::to_char,
+                         [&stream] (char ch) { stream << ch; });
+
+        auto stream_f = [&] (auto const & previous_seq, auto const & aligned_seq)
+        {
+            // write alignment bars
+            stream << '\n' << std::setw(8) << "";
+            std::ranges::for_each(views::zip(previous_seq, aligned_seq) | views::slice(begin_pos, end_pos),
+                                  [&stream] (auto && ch) { stream << (get<0>(ch) == get<1>(ch) ? '|' : ' '); });
+
+            // write next sequence
+            stream << '\n' << std::setw(8) << "";
+            std::ranges::for_each(aligned_seq | views::slice(begin_pos, end_pos) | views::to_char,
+                                  [&stream] (char ch) { stream << ch; });
+        };
+        (stream_f(get<idx>(align), get<idx + 1>(align)), ...);
+        stream << '\n';
+    }
+}
+} // namespace seqan3::detail
+
+namespace seqan3
+{
+/*!\brief Stream operator for alignments, which are represented as tuples of aligned sequences.
+ * \ingroup aligned_sequence
+ *
+ * \tparam alignment_t The alignment type, must satisfy tuple_like and its size must be at least 2.
+ *
+ * \param[in,out] stream The target stream for the formatted output.
+ * \param[in] alignment The alignment that shall be formatted. All sequences must be equally long.
+ *
+ * \return The given stream to which the alignment representation is appended.
+ */
+template <typename char_t, typename alignment_t>
+//!\cond
+    requires (detail::debug_streamable_tuple<alignment_t> &&
+              detail::all_satisfy_aligned_seq<detail::alignment_type_list_t<remove_cvref_t<alignment_t>>>)
```

Comment:
`alignment_type_list_t` is unknown, did you forget to add something?

Reply:
Changing this to 

```suggestion
              detail::all_satisfy_aligned_seq<detail::tuple_type_list_t<remove_cvref_t<alignment_t>>>)
```

Did not compile for me on gcc9 and gcc10.